src/components/CopyableWrapper.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <div class="copyable-wrapper">
+    <slot />
+    <PButton size="sm" class="copyable-wrapper__button" @click="copy">
+      Copy <p-icon icon="DuplicateIcon" />
+    </PButton>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { PButton } from '@prefecthq/prefect-design'
+  import { copyToClipboard } from '@/utilities/copy'
+
+  const props = defineProps<{
+    textToCopy: string,
+  }>()
+
+  function copy(): void {
+    copyToClipboard(props.textToCopy)
+  }
+</script>
+
+<style>
+.copyable-wrapper { @apply
+  relative
+  min-h-[32px]
+}
+
+.copyable-wrapper:hover .copyable-wrapper__button { @apply
+  opacity-100
+}
+
+.copyable-wrapper__button { @apply
+  absolute
+  top-0
+  right-0
```

Comment:
maybe `top/right-1` or even 2 to give it just a touch of breathing room from the edge?

Reply:
I thought of that, but ultimately went with the assumption that most of these are in p-card, which has plenty of padding. I don't want to put a lot of opinions in a generic component that's meant to just wrap your content